dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/spirv/ProgramBuilderSpv.java

```diff
@@ -381,7 +396,8 @@ private Thread createThreadFromFunction(Function function, int x, int y, int z)
         Memory memory = program.getMemory();
 
         // Create thread-local variables
-        BuiltIn builtIn = (BuiltIn) getDecoration(DecorationType.BUILT_IN);
+        BuiltIn builtIn = (BuiltIn) getDecoration(BUILT_IN);
+        builtIn.setDimension(threadGrid.get(0), threadGrid.get(1), 1);
```

Comment:
Why is the last parameter always 1?

Reply:
I understand our config as:

- `threadGrid.get(0)`: thread number of a subworkgroup
- `threadGrid.get(1)`: subworkgroup number of a workgroup
- `threadGrid.get(2)`: workgroup number of a shader

I think this is also consistent with we using them [here](https://github.com/hernanponcedeleon/Dat3M/blob/spirv-visitor/dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/spirv/ProgramBuilderSpv.java#L378). I used 1 here for the missing dimension of WorkgroupSize.